src/java.base/share/classes/sun/net/www/protocol/mailto/Handler.java

```diff
@@ -126,33 +122,16 @@ public void parseURL(URL u, String spec, int start, int limit) {
         /*
          * Let's just make sure we DO have an Email address in the URL.
          */
-        boolean nogood = false;
-        if (file == null || file.isEmpty())
-            nogood = true;
-        else {
-            boolean allwhites = true;
-            for (int i = 0; i < file.length(); i++)
-                if (!Character.isWhitespace(file.charAt(i)))
-                    allwhites = false;
-            if (allwhites)
-                nogood = true;
-        }
-        if (nogood)
+        if (file.isBlank())
```

Comment:
The mail protocol handler is a legacy protocol handler and I don't know if there is good test coverage. The change proposed here changes the exception from RuntimeException to NPE when there isn't a file component in the URL. Neither is specified by URLStreamHandler.parseURL so I suspect there may be follow-on issues to clarify the spec on several points.

Reply:
`file` can't be `null` here.  It's either empty string, or result of String.substring (which is also guarantee to be non-null).
Check code before this line:
https://github.com/openjdk/jdk/blob/07669e3bc65b1728d784e21ec83b437374f9fa19/src/java.base/share/classes/sun/net/www/protocol/mailto/Handler.java#L121-L125